my_enumerables.rb

```diff
@@ -0,0 +1,190 @@
+
+    arr = to_a
+    arr2 = []
+    arr.my_each_with_index do |_item, index|
+      arr2.push(arr[index]) if yield arr[index]
+    end
+    arr2
+  end
+
+  def my_all?(args = nil)
+    return to_enum unless block_given? || !args.nil?
+
+    arr = to_a
+    if block_given?
+      arr.my_each_with_index do |_item, index|
+        return false unless yield arr[index]
+      end
+    elsif args.is_a? Class
+      arr.my_each_with_index do |item, _index|
+        return false unless item.class.ancestors.include?(args)
+      end
+    elsif args.is_a? Regexp
+      arr.my_each_with_index do |item, _index|
+        return false unless item.match(args)
+      end
+    end
+    true
+  end
+
+  def my_any?(args = nil)
+    return to_enum unless block_given? || !args.nil?
+
+    arr = to_a
+    any = false
+
+    if block_given?
+      arr.my_each_with_index do |_item, index|
+        any = true if yield arr[index]
+      end
+    elsif args.is_a? Class
+      arr.my_each_with_index do |item, _index|
+        puts item.class.ancestors
+        any = true if item.class.ancestors.include?(args)
+      end
+    elsif args.is_a? Regexp
+      arr.my_each_with_index do |item, _index|
+        any = true if item.match(args)
+      end
+    end
+    any
+  end
+
+  def my_none?(args = nil)
+    return to_enum unless block_given? || !args.nil?
+
+    arr = to_a
+    if block_given?
+      arr.my_each_with_index do |_item, index|
+        return true unless yield arr[index]
+      end
+    elsif args.is_a? Class
+      arr.my_each_with_index do |item, _index|
+        return true unless item.class.ancestors.include?(args)
+      end
+    elsif args.is_a? Regexp
+      arr.my_each_with_index do |item, _index|
+        return true unless item.match(args)
+      end
+    end
+    false
+  end
+
+  def my_count(arg = nil)
+    index = 0
+    count = 0
+
+    if arg
+      while index < length
+        count += 1 if proc.call self[index]
+        index += 1
+      end
+    elsif block_given?
+      while index < length
+        count += 1 if yield self[index]
+        index += 1
+      end
+    else
+      my_each do |_item|
+        count += 1
+        index += 1
+      end
+    end
+    count
+  end
+
+  def my_map(pro = nil)
+    return to_enum unless block_given?
+
+    arr = to_a
+    arr2 = []
+    if pro
+      arr.my_each_with_index do |_item, index|
+        arr2.push(proc.cal(arr[index]))
+      end
+    else
+      arr.my_each_with_index do |_item, index|
+        arr2.push(yield arr[index])
+      end
+    end
+    arr2
+  end
+
+  def my_inject(arg = self[0], sym = nil)
+    arr = to_a
+    i = 1
+    result = arg
+    if arg.is_a? Symbol
+      sym = arg
+      result = arr[0]
+    end
+    if sym
+      i = 0
+      while i < arr.length
+        result = result.send sym, arr[i]
+        i += 1
+      end
+      result
+    end
+    while i < arr.length
+      result = yield result, arr[i]
+      i += 1
+    end
+    result
+  end
```

Comment:
#my_inject when a block is given without an initial value combines all elements of enum by applying a binary operation, specified by a block::range
#my_inject when a block is given with an initial value combines all elements of enum by applying a binary operation, specified by a block
#my_inject when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method
#my_inject when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method:range